public/app/plugins/datasource/loki/types.ts

```diff
@@ -35,20 +35,13 @@ export enum LokiQueryDirection {
   Forward = 'forward',
 }
 
-export interface LokiQuery extends DataQuery {
-  queryType?: LokiQueryType;
-  expr: string;
+export interface LokiQuery extends Omit<LokiQueryFromSchema, 'queryType'> {
   direction?: LokiQueryDirection;
-  legendFormat?: string;
-  maxLines?: number;
-  resolution?: number;
-  /** Used in range queries */
   volumeQuery?: boolean;
-  /* @deprecated now use queryType */
-  range?: boolean;
-  /* @deprecated now use queryType */
-  instant?: boolean;
-  editorMode?: QueryEditorMode;
+  // CUE autogenerates `queryType` as `?string`, as that's how it is defined
+  // in the parent-interface (in DataQuery).
+  // to fix it, we override it here.
+  queryType?: LokiQueryType;
```

Comment:
`LokiQueryType` should probably also be moved to the cue file and then exported from there.

Reply:
@Clarity-89 sorry, could you explain this in more detail? nothing in the CUE-file will refer to `LokiQueryType`.. so, why is it good to have it there?

Reply:
When we fix the bug with `queryType` not resolving correctly, it'll be set as `LokiQueryType` in cue. But even if it's not used in cue it's better still to define it there and then import it from the generated file, so we have it schematised.

Reply:
i see. done.